test/test_eos_unit.cpp

```diff
@@ -131,6 +132,13 @@ inline void compare_two_eoss(E1 &&test_e, E2 &&ref_e) {
   return;
 }
 
+SCENARIO("Test that we can either throw an error on host or do nothing on device",
+         "[RequireMaybe]") {
+  // TODO(JMM): For whatever reason, the preprocessor does not like it if I
+  // call `PORTABLE_ALWAYS_THROW_OR_ABORT
+  REQUIRE_MAYBE_THROWS(PORTABLE_ALWAYS_THROW_OR_ABORT("Error message"));
```

Comment:
Is catch ok when this scenario is empty as is the case if portability strategy none is not defined @Yurlungur ?

Reply:
Catch is ok with empty scenarios I think. The scenario just auto-passes with zero conditions I believe.

Reply:
Is the empty scenario scenario tested in our CI @Yurlungur ?

Reply:
Yes if the CI passes on GPU it passed for the empty scenario.